### ISSUE TITLE
Crash when Bors attempts to push into externally updated target branch

### ISSUE THREAD

**Issue by noizwaves** [bors-ng/bors-ng#856]
When bors completes a batch, it pushes `batch.commit` onto `batch.into_branch` (https://github.com/bors-ng/bors-ng/blob/master/lib/worker/batcher.ex#L528). This call can fail if the `into_branch` was updated after the Batch's staging branch was created. This scenario will crash Bors.

In our usage of Bors, we will often need to fix high priority issues by pushing directly to the master (thereby side-stepping the merge queue).

The error message returned by GitHub includes a JSON encoded body with message that indicates whether the failure is due to a non-fast forward merge (`Update is not a fast forward`), so this could be used when deciding how to handle this failure.

This is my first time submitting an issue here, so please do let me know if I've missed any info in this. I'm also investigating what a fix for this would look like, so any input you have would be appreciated!

## Stack trace
Here's an old stack trace from such a crash.
```
{{:badmatch,
  {:error, :push, 422,
   "{\"message\":\"Update is not a fast forward\",\"documentation_url\":\"https://developer.github.com/v3/git/refs/#update-a-reference\"}"}},
 [
   {BorsNG.Worker.Batcher, :complete_batch, 3,
    [file: 'lib/worker/batcher.ex', line: 518]},
   {BorsNG.Worker.Batcher, :maybe_complete_batch, 1,
    [file: 'lib/worker/batcher.ex', line: 499]},
   {BorsNG.Worker.Batcher, :handle_cast, 2,
    [file: 'lib/worker/batcher.ex', line: 84]},
   {:gen_server, :try_dispatch, 4,
    [file: 'gen_server.erl', line: 637]},
   {:gen_server, :handle_msg, 6,
    [file: 'gen_server.erl', line: 711]},
   {:proc_lib, :init_p_do_apply, 3,
    [file: 'proc_lib.erl', line: 249]}
 ]}
```

## Steps to reproduce
1. Create a PR
1. Have the PR approved
1. Add comment `bors merge` to the PR
1. Shortly after the batch starts running, push a commit directly to master
1. When the build is finishing, bors crashes

## Actual behavior
Bors crashes

## Expected behavior
Bors should to retry the batch immediately. There is no evidence that the failure is due to a bad PR, so the batch should not be bisected.

**Comment by cryslith** [comment]
This was fixed in #859, right?

**Comment by noizwaves** [comment]
@cryslith correct, closing this now
